### ISSUE TITLE
Add Record Method

### ISSUE THREAD

**Issue by CodeDarigan** [CodeDarigan/WAT#121]
Some properties change over time. A Recorder would take an object and then users would be able to set which properties they want to record. Recorder would check the properties each process loop and add it to a collection of recorded elements

```
var recorder = record(Player)
recorder.record("height")
recorder.record("health")

// Implementation
func _process(delta) -> void:
    recorded.append(element.get("property"))
````

We could add a double version where we can actually extend it to be triggered on setget (maybe something like "add setter collection")

**Comment by CodeDarigan** [comment]
We can't add setget due to how we create doubles (we extend what we are doubling and you can't rewrite a predefined variable)

**Comment by CodeDarigan** [comment]
Resolved by e4cdc8b94819e175b6f1846823506593fec218eb

**Comment by CodeDarigan** [comment]
Note - Only basic version implemented. Double has not been implemented.
